src/plots/smith/smith.js

```diff
@@ -0,0 +1,1456 @@
+        _this.angularTickLayout = newTickLayout;
+    }
+
+    var vals = Axes.calcTicks(ax);
+
+    // angle of polygon vertices in geometric radians (null means circles)
+    // TODO what to do when ax.period > ax._categories ??
+    var vangles;
+    if(smithLayout.gridshape === 'linear') {
+        vangles = vals.map(t2g);
+
+        // ax._vals should be always ordered, make them
+        // always turn counterclockwise for convenience here
+        if(Lib.angleDelta(vangles[0], vangles[1]) < 0) {
+            vangles = vangles.slice().reverse();
+        }
+    } else {
+        vangles = null;
+    }
+    _this.vangles = vangles;
+
+    // Use tickval filter for category axes instead of tweaking
+    // the range w.r.t sector, so that sectors that cross 360 can
+    // show all their ticks.
+    if(ax.type === 'category') {
+        vals = vals.filter(function(d) {
+            return Lib.isAngleInsideSector(t2g(d), _this.sectorInRad);
+        });
+    }
+
+    if(ax.visible) {
+        Axes.drawGrid(gd, ax, {
+            // vals: vals,
+            vals: [-5.0, -2.0, -1.0, -0.5, -0.2, 0.2, 0.5, 1.0, 2.0, 5.0].map(function(v) {
+                return {
+                    x: v,
+                };
+            }),
+            layer: layers['angular-grid'],
+            path: gridPathFn,
+            transFn: Lib.noop,
+            crisp: false
+        });
+
+        Axes.drawLabels(gd, ax, {
+            vals: [-5.0, -2.0, -1.0, -0.5, -0.2, 0.0, 0.2, 0.5, 1.0, 2.0, 5.0, Infinity].map(function(v) {
+                var radius = _this.radius / v;
+                var theta = 2.0 * Math.atan2(radius, _this.radius);
+
+                return {
+                    x: theta * 360.0 / (2 * Math.PI),
+                    text: v === Infinity ? '∞' : ((v === 0.0 ? '0 + 0' : v) + 'j'),
+                    font: '"Open Sans", verdana, arial, sans-serif',
+                    fontColor: '#444',
+                    fontSize: v === Infinity ? 16 : 12,
+                };
+            }),
+            layer: layers['angular-axis'],
+            repositionOnUpdate: true,
+            transFn: transFn,
+            labelFns: labelFns
+        });
+    }
+
+    // TODO maybe two arcs is better here?
+    // maybe split style attributes between inner and outer angular axes?
+
+    updateElement(layers['angular-line'].select('path'), angularLayout.showline, {
+        d: _this.pathSubplot(),
+        transform: strTranslate(cx, cy)
+    })
+    .attr('stroke-width', angularLayout.linewidth)
+    .call(Color.stroke, angularLayout.linecolor);
+};
+
+proto.updateFx = function(fullLayout, smithLayout) {
+    if(!this.gd._context.staticPlot) {
+        this.updateAngularDrag(fullLayout);
+        this.updateRadialDrag(fullLayout, smithLayout, 0);
+        this.updateRadialDrag(fullLayout, smithLayout, 1);
+        this.updateMainDrag(fullLayout);
+    }
+};
+
+proto.updateMainDrag = function(fullLayout) {
```

Comment:
Do we need this big block of code?

Reply:
Some of it is necessary in order for the hover action to work, but I'm not sure how to evaluate what to cut out of here.

Reply:
I still think you could simply remove the rest.

Reply:
See 6b0f08f867fc70bd17a4c705c9c8b64ca74ecb3a--much of this function is now gone.